### ISSUE TITLE
Question about the implementation of normalization stat

### ISSUE THREAD

**Issue by wswdx** [bytetriper/RAE#51]
Thanks for your great work!
I have a question about the implementation of pre-decode normalization: 
Since the last layer of the encoder in the ViT architecture always includes LayerNorm, and in your code, the scaling and shifting of this LayerNorm have been removed, then it means that each token of each sample already has zero mean and unit variance. Why do we still need to separately compute the mean and variance of the entire training set? Theoretically, shouldn't the entire dataset also have zero mean and unit variance? However, based on the stat.pt file you provided, it doesn't seem to be the case.

**Comment by bytetriper** [comment]
Hi,

LayerNorm does not guarantee unit variance of the underlying distribution; it only enforces that each individual vector has unit variance (i.e., tensor.std() equals 1). Therefore, a BatchNorm computed over the training set is still required to ensure unit variance at the distribution level.
